lib/ansible/module_utils/cisco_ucs.py

```
@@ -0,0 +1,94 @@
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+
+try:
+    import ucsmsdk
+    HAS_UCSMSDK = True
+except:
+    HAS_UCSMSDK = False
+
+try:
+    import ucsm_apis
+    HAS_UCSMAPI = True
+except:
+    HAS_UCSMAPI = False
+
+
+class UcsConnection():
+
+    @staticmethod
+    def is_login_param(param):
+        return param in ["ucs_ip", "ucs_username", "ucs_password",
+                         "ucs_port", "ucs_secure", "ucs_proxy", "ucs_server"]
+
+    def __init__(self, module):
+        if HAS_UCSMSDK is False:
+            results = {}
+            results["msg"] = "ucsmsdk is not installed"
+            module.fail_json(**results)
+        if HAS_UCSMAPI is False:
+            results = {}
+            results["msg"] = "ucsm_apis is not installed"
+            module.fail_json(**results)
+        self.module = module
+        self.handle = None
+
+    def login(self):
+        ansible = self.module.params
+        server = ansible.get('ucs_server')
+        if server:
+            return server
+
+        from ucsmsdk.ucshandle import UcsHandle
+        results = {}
+        try:
+            server = UcsHandle(ip=ansible["ucs_ip"],
+                               username=ansible["ucs_username"],
+                               password=ansible["ucs_password"],
+                               port=ansible["ucs_port"],
+                               secure=ansible["ucs_secure"],
+                               proxy=ansible["ucs_proxy"])
+            server.login()
+        except Exception as e:
```

IMHO not a blocker but try to avoid catch all exception handling

---

@resmo - This will require more structural changes at SDK level which is being internally consumed by this module. We will take this as TO DO task to be done in next SDK release and will fix this review comment in another PR later.